internal/mode/static/manager.go

```diff
@@ -275,3 +289,26 @@ func setInitialConfig(
 	// resource is processed by the controller
 	return updateControlPlane(&config, logger, eventRecorder, configName, logLevelSetter)
 }
+
+func configureNginxMetrics(gatewayClassName string) error {
+	constLabels := map[string]string{"class": gatewayClassName}
+	ngxCollector, err := nkgMetrics.NewNginxMetricsClient(constLabels)
+	if err != nil {
+		return fmt.Errorf("cannot get NGINX metrics: %w", err)
+	}
```

Comment:
what will happen if NGINX stops responding to the Prometheus  collector? Will that error appear anywhere? Will NKG continue to work?

Reply:
I'm not sure. Do you know how I could test this?

Reply:
> I'm not sure. Do you know how I could test this?

We should be able to get shell in the NGINX container, update NGINX config to remove the unix socket listen and reload NGINX

Reply:
Opened https://github.com/nginxinc/nginx-prometheus-exporter/issues/484